src/tests/SensorTest.cpp

```diff
@@ -19,84 +19,159 @@ struct SensorTest : Cr::TestSuite::Tester {
+  CORRADE_VERIFY(parentNode.getSubtreeSensorSuite().getSensors().size() == 5);
+  CORRADE_VERIFY(childNode.getNodeSensorSuite().getSensors().size() == 3);
+  CORRADE_VERIFY(childNode.getSubtreeSensorSuite().getSensors().size() == 3);
+
+  // Add sensors to grandchild node
+  auto sensorSpec3A = CameraSensorSpec::create();
+  sensorSpec3A->uuid = "3A";
+  auto sensorSpec3B = CameraSensorSpec::create();
+  sensorSpec3B->uuid = "3B";
+  auto sensorSpec3C = CameraSensorSpec::create();
+  sensorSpec3C->uuid = "3C";
+  auto sensorSpec3D = CameraSensorSpec::create();
+  sensorSpec3D->uuid = "3D";
+  SensorFactory::createSensors(
+      grandchildNode, {sensorSpec3A, sensorSpec3B, sensorSpec3C, sensorSpec3D});
+  CORRADE_VERIFY(parentNode.getNodeSensorSuite().getSensors().size() == 2);
+  CORRADE_VERIFY(parentNode.getSubtreeSensorSuite().getSensors().size() == 9);
+  CORRADE_VERIFY(childNode.getNodeSensorSuite().getSensors().size() == 3);
+  CORRADE_VERIFY(childNode.getSubtreeSensorSuite().getSensors().size() == 7);
+  CORRADE_VERIFY(grandchildNode.getNodeSensorSuite().getSensors().size() == 4);
+  CORRADE_VERIFY(grandchildNode.getSubtreeSensorSuite().getSensors().size() ==
+                 4);
+
+  // Remove sensor from parentNode
+  SensorFactory::deleteSensor(parentNode, "1A");
+  CORRADE_VERIFY(parentNode.getNodeSensorSuite().getSensors().size() == 1);
+  CORRADE_VERIFY(parentNode.getSubtreeSensorSuite().getSensors().size() == 8);
+  CORRADE_VERIFY(childNode.getNodeSensorSuite().getSensors().size() == 3);
+  CORRADE_VERIFY(childNode.getSubtreeSensorSuite().getSensors().size() == 7);
+  CORRADE_VERIFY(grandchildNode.getNodeSensorSuite().getSensors().size() == 4);
+  CORRADE_VERIFY(grandchildNode.getSubtreeSensorSuite().getSensors().size() ==
+                 4);
+
+  // Remove sensor from child node
+  SensorFactory::deleteSensor(parentNode, "2A");
+  CORRADE_VERIFY(parentNode.getNodeSensorSuite().getSensors().size() == 1);
+  CORRADE_VERIFY(parentNode.getSubtreeSensorSuite().getSensors().size() == 7);
+  CORRADE_VERIFY(childNode.getNodeSensorSuite().getSensors().size() == 2);
+  CORRADE_VERIFY(childNode.getSubtreeSensorSuite().getSensors().size() == 6);
+  CORRADE_VERIFY(grandchildNode.getNodeSensorSuite().getSensors().size() == 4);
+  CORRADE_VERIFY(grandchildNode.getSubtreeSensorSuite().getSensors().size() ==
+                 4);
+
+  // Remove sensor from grandchild node
+  SensorFactory::deleteSensor(parentNode, "3A");
+  CORRADE_VERIFY(parentNode.getNodeSensorSuite().getSensors().size() == 1);
+  CORRADE_VERIFY(parentNode.getSubtreeSensorSuite().getSensors().size() == 6);
+  CORRADE_VERIFY(childNode.getNodeSensorSuite().getSensors().size() == 2);
+  CORRADE_VERIFY(childNode.getSubtreeSensorSuite().getSensors().size() == 5);
+  CORRADE_VERIFY(grandchildNode.getNodeSensorSuite().getSensors().size() == 3);
+  CORRADE_VERIFY(grandchildNode.getSubtreeSensorSuite().getSensors().size() ==
+                 3);
+
```

Comment:
You can try to delete a sensor that never exists.

Reply:
Sounds good, I can catch the exception.